app/src/main/java/com/powerhungers/fomezero/feature/registration/domain/usecase/RegistrationUseCase.kt

```diff
@@ -0,0 +1,21 @@
+package com.powerhungers.fomezero.feature.registration.domain.usecase
+
+import com.powerhungers.fomezero.common.exception.MissInformationException
+import com.powerhungers.fomezero.feature.login.domain.model.User
+import com.powerhungers.fomezero.feature.registration.domain.model.Register
+import com.powerhungers.fomezero.feature.registration.domain.repository.RegistrationRepository
+import kotlinx.coroutines.flow.flow
+import java.util.concurrent.Flow
+
+class RegistrationUseCase(private val repository: RegistrationRepository) {
+    operator fun invoke(
+        name: String, userType: String,
+        email: String, password: String): Flow<User> = when {
+        name.isBlank() -> flow { throw MissInformationException() }
+        userType.isBlank() -> flow { throw MissInformationException() }
+        email.isBlank() -> flow { throw MissInformationException() }
+        password.isBlank() -> flow { throw MissInformationException() }
+        else -> repository.registration(name, userType, email, password)
+
+    }
```

Comment:
```suggestion
    operator fun invoke(name: String, userType: String, email: String, password: String): Flow<User> = when {
        name.isBlank() -> flow { throw MissInformationException() }
        userType.isBlank() -> flow { throw MissInformationException() }
        email.isBlank() -> flow { throw MissInformationException() }
        password.isBlank() -> flow { throw MissInformationException() }
        else -> repository.registration(name, userType, email, password)
    }
```
a identação desse metodo estava dificultando a leitura, nao estava claro o que estava dentro do when e nem o que eram parametros do metodo pois estavam todos alinhados na mesma linha

Reply:
o que acha de criar exceptions mais especificas para poder informar ao usuario qual campo exatamente que esta vazio? dessa forma nao temos como saber qual campo esta vazio